src/CTA.WebForms2Blazor/ClassConverters/ControlCodeBehindClassConverter.cs

```diff
@@ -1,24 +1,52 @@
-﻿using System;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
 using CTA.WebForms2Blazor.FileInformationModel;
+using CTA.WebForms2Blazor.Helpers;
+using CTA.WebForms2Blazor.Extensions;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Linq;
+using System.Text;
+using System.IO;
 
 namespace CTA.WebForms2Blazor.ClassConverters
 {
     public class ControlCodeBehindClassConverter : ClassConverter
     {
         public ControlCodeBehindClassConverter(
             string relativePath,
+            string sourceProjectPath,
             SemanticModel sourceFileSemanticModel,
+            TypeDeclarationSyntax originalDeclarationSyntax,
             INamedTypeSymbol originalClassSymbol)
-            : base(relativePath, sourceFileSemanticModel, originalClassSymbol)
+            : base(relativePath, sourceProjectPath, sourceFileSemanticModel, originalDeclarationSyntax, originalClassSymbol)
         {
-
+            // TODO: Register with the necessary services
         }
 
         public override async Task<FileInformation> MigrateClassAsync()
         {
-            throw new NotImplementedException();
+            // NOTE: For now we make no code modifications, just to be
+            // ready for the demo and produces files
+            var requiredNamespaces = _sourceFileSemanticModel.GetNamespacesReferencedByType(_originalDeclarationSyntax);
+            var usingStatements = CodeSyntaxHelper.BuildUsingStatements(requiredNamespaces.Select(namespaceSymbol => namespaceSymbol.Name));
+            var namespaceNode = CodeSyntaxHelper.BuildNamespace(_originalClassSymbol.ContainingNamespace.Name, _originalDeclarationSyntax);
+            var fileText = CodeSyntaxHelper.GetFileSyntaxAsString(namespaceNode, usingStatements);
+
+            return new FileInformation(GetNewRelativePath(), Encoding.UTF8.GetBytes(fileText));
+        }
+
+        private string GetNewRelativePath()
+        {
+            // TODO: Potentially remove certain folders from beginning of relative path
+            var relativePathFolder = Path.GetDirectoryName(_relativePath);
+            var oldFileName = Path.GetFileName(_relativePath);
+            // Path.GetFileNameWithoutExtension only removes the .cs of .ascx.cs
+            // and so we have to remove the extension manually
+            var baseFileNameLength = oldFileName.Length - Constants.ControlCodeBehindExtension.Length;
+            // The code behinds are responsible for generating the new razor
+            // file and so we replace the extension with .razor
+            var newFileName = oldFileName.Substring(0, baseFileNameLength) + Constants.RazorFileExtension;
+            return Path.Combine(Constants.RazorComponentDirectoryName, relativePathFolder, newFileName);
```

Comment:
A couple of things:

1. Since this logic is used in multiple places, consider creating a private method (or static utility method) to replace the extension.
2. Not sure if you're aware, but similar to WebForms, Blazor can also have a code-behind files (i.e. `.razor` for front-end and `.razor.cs` for code-behind). Not sure if this is something you've considered or if it fits into your design, but here is a [brief example](https://www.claudiobernasconi.ch/2020/02/08/blazor-components-how-to-split-template-and-csharp-code/) if you're unfamiliar.